api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationSearchAdviserTest.java

```diff
@@ -0,0 +1,446 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.locationbasedaccess.aop;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.GlobalProperty;
+import org.openmrs.Location;
+import org.openmrs.LocationTag;
+import org.openmrs.User;
+import org.openmrs.api.LocationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.locationbasedaccess.aop.common.AOPContextSensitiveTest;
+import org.openmrs.module.locationbasedaccess.aop.common.TestWithAOP;
+import org.openmrs.module.locationbasedaccess.aop.interceptor.LocationServiceInterceptorAdvice;
+import org.openmrs.util.OpenmrsConstants;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class LocationSearchAdviserTest extends AOPContextSensitiveTest {
+
+    private LocationService locationService;
+
+    private static final String LOCATION_TEST_DATA_XML_LOCATION = "include/LocationTestData.xml";
+
+    private static final String location1Uuid = "ef93c695-ac43-450a-93f8-4b2b4d50a3c8";
+    private static final String location2Uuid = "ef93c695-ac43-450a-93f8-4b2b4d50a3c9";
+    private static final String location3Uuid = "ef93c695-ac43-450a-93f8-4b2b4d50a3ca";
+    private static final String location4Uuid = "ef93c695-ac43-450a-93f8-4b2b4d50a3cb";
+    private static final String location5Uuid = "ef93c695-ac43-450a-93f8-4b2b4d50a3cc";
+    private static final String badLocationUuid = "badc0de1-cafe-d00d-0123-456789abcdef";
+
+    private static final String location2Name = "Test Location 2";
+    private static final String location3Name = "Test Location 3";
+
+    private static final String searchQuery = "test location";
+    private static final String tagName = "Test Location Tag";
+
+    private Set<String> restrictedGetMethodNames = new HashSet<String>();
+    private Set<String> allLocationUuids = new HashSet<String>();
+    private Set<String> unretiredLocationUuids = new HashSet<String>();
+    private Set<String> rootLocationUuids = new HashSet<String>();
+    private Set<String> locationsWithTagUuids = new HashSet<String>();
```

Comment:
Why you have created so many set<> here? Can you reduce this if it used in 1-2 test cases? 
Otherwise it's hard to understand actual use cases of the test cases.